src/main/integrations/slack/index.ts

```diff
@@ -0,0 +1,81 @@
+import playerStateStore, { PlayerState, VideoDetails } from "../../player-state-store";
+import IIntegration from "../integration";
+import { StoreSchema } from "~shared/store/schema";
+import Conf from "conf";
+
+export default class SlackStatus implements IIntegration {
+  private store: Conf<StoreSchema>;
+  private enabled = false;
+  private stateCallback: (event: PlayerState) => void = null;
+  private currentVideoId: string = "";
+  private statusTimeoutId: NodeJS.Timeout;
+  private slackUserToken: string = "";
+
+  private async playerStateChanged(state: PlayerState) {
+    const { secondsTillClearStatus } = await this.store.get("slack");
+    if (state.trackState === 1 && state.videoDetails) {
+      if (!state.adPlaying) {
+        if (this.statusTimeoutId) clearTimeout(this.statusTimeoutId);
+        if (this.currentVideoId === state.videoDetails.id) {
+          this.statusTimeoutId = setTimeout(() => {
+            this.currentVideoId = "";
+            this.updateSlackStatus(true, state.videoDetails);
+          }, secondsTillClearStatus * 1000);
+        } else {
+          this.currentVideoId = state.videoDetails.id;
+          this.updateSlackStatus(false, state.videoDetails);
+        }
+      }
+    }
+  }
+  private updateSlackStatus(clear: boolean, data: VideoDetails) {
+    const message = `Listening To: ${data.title} | ${data.author}`;
+    let postData: { status_text: string; status_emoji: string; status_expiration: number } = {
+      status_text: "",
+      status_emoji: "",
```

Comment:
*Idea*
🎵 Emoji?

Reply:
Not a bad idea. I could have a box where you can enter the emoji you wish to use or offer a few options.